### ISSUE TITLE
cygwin: t/test_alien.t locks up with fork() error

### ISSUE THREAD

**Issue by plicease** [Perl5-Alien/Alien-Build#153]
I think this used to work on Cygwin and it is possible that it is just me, but the test loops on a fork() error.

```
% prove -lvm t/test_alien.t 
t/test_alien.t .. 
ok 1 - alien_ok {
    ok 1 - as class {
        ok 1 - alien_ok with class
        ok 2 - bin_dir added to path
        1..2
    }
    ok 2 - as object {
        ok 1 - alien_ok with class
        ok 2 - bin_dir added to path
        1..2
    }
    ok 3 - alien_ok with instance
    ok 4 - alien_ok with bad class
    ok 5 - alien_ok with undef
    1..5
}
ok 2 - helper_ok {
    ok 1 - Alien::libfoo1 responds to: cflags libs dynamic_libs bin_dir
    ok 2 - helper foo1 exists
    ok 3 - something else
    ok 4 - default test name
    ok 5 - custom name
    ok 6 - failed test
    1..6
}
ok 3 - interpolate_template_is {
    ok 1 - Alien::libfoo1 responds to: cflags libs dynamic_libs bin_dir
    ok 2 - template matches
    ok 3 - template matches
    ok 4 - pass with default name
    ok 5 - pass with custom name
    ok 6 - fail with string match
    ok 7 - fail with string match
    ok 8 - bogus helper
    1..8
}
ok 4 - ffi_ok {
    1..0 # SKIP Test requires FFI::Platypus
}
      1 [main] perl 505 child_info_fork::abort: unable to map \??\D:\lang\cygwin\64\home\ollisg\dev\Alien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll (using D:\lang\cygwin\64\home\ollisg\dev\Al
ien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126      1 [main] perl 505 child_info_fork::abort: unable to map \??\D:\lang\cygwin\64\home\ollisg\dev\Alien-Build\.tmp\test-alien-
ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll (using D:\lang\cygwin\64\home\ollisg\dev\Alien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126

      0 [main] perl 506 child_info_fork::abort: unable to map \??\D:\lang\cygwin\64\home\ollisg\dev\Alien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll (using D:\lang\cygwin\64\home\ollisg\dev\Al
ien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      0 [main] perl 506 child_info_fork::abort: unable to map \??\D:\lang\cygwin\64\home\ollisg\dev\Alien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll (using D:\lang\cygwin\64\home\ollisg\dev\Al
ien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      0 [main] perl 507 child_info_fork::abort: unable to map \??\D:\lang\cygwin\64\home\ollisg\dev\Alien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll (using D:\lang\cygwin\64\home\ollisg\dev\Al
ien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      0 [main] perl 507 child_info_fork::abort: unable to map \??\D:\lang\cygwin\64\home\ollisg\dev\Alien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll (using D:\lang\cygwin\64\home\ollisg\dev\Al
ien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      1 [main] perl 508 child_info_fork::abort: unable to map \??\D:\lang\cygwin\64\home\ollisg\dev\Alien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll (using D:\lang\cygwin\64\home\ollisg\dev\Al
ien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      1 [main] perl 508 child_info_fork::abort: unable to map \??\D:\lang\cygwin\64\home\ollisg\dev\Alien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll (using D:\lang\cygwin\64\home\ollisg\dev\Al
ien-Build\.tmp\test-alien-ugofii\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126

```

I did try rebasing the generated .dll but that didn't seem to help anything.

```diff
diff --git a/lib/Test/Alien.pm b/lib/Test/Alien.pm
index 06f0a75..94d87c4 100644
--- a/lib/Test/Alien.pm
+++ b/lib/Test/Alien.pm
@@ -618,6 +618,21 @@ sub xs_ok
         $libpath->parent->mkpath;
         move($lib, "$libpath") || die "unable to copy $lib => $libpath $!";

+        if($^O eq 'cygwin')
+        {
+          my($out, $ok) = capture_merged {
+            my @cmd = ('/usr/bin/rebase', '-sOT', $libpath);
+            print "+@cmd\n";
+            system @cmd;
+            not $?;
+          };
+          unless($ok)
+          {
+            warn $out;
+            die "cygwin rebase failed";
+          }
+        }
+
         pop @modparts;
         my $pmpath = path($dir)->child(@modparts, "$modfname.pm");
         $pmpath->parent->mkpath;
```

**Comment by coynesm** [comment]
I have had a similar issue installing Alien::Build via cpan:

--> Working on Alien::Build
Fetching http://www.cpan.org/authors/id/P/PL/PLICEASE/Alien-Build-2.17.tar.gz ... OK
Configuring Alien-Build-2.17 ... OK
Building and testing Alien-Build-2.17 ...       1 [main] perl 9869 child_info_fork::abort: un
able to map \??\C:\cygwin64\home\username\.cpanm\work\1585951500.9548\Alien-Build-2.17\.tmp\test
-alien-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0.dll (using C:\cygwin64\home\username\.cpanm\work\1585
951500.9548\Alien-Build-2.17\.tmp\test-alien-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32
error 126
      0 [main] perl 9870 child_info_fork::abort: unable to map \??\C:\cygwin64\home\username\.cp
anm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-alien-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0
.dll (using C:\cygwin64\home\username\.cpanm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-ali
en-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      0 [main] perl 9871 child_info_fork::abort: unable to map \??\C:\cygwin64\home\username\.cp
anm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-alien-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0
.dll (using C:\cygwin64\home\username\.cpanm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-ali
en-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      0 [main] perl 9872 child_info_fork::abort: unable to map \??\C:\cygwin64\home\username\.cp
anm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-alien-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0
.dll (using C:\cygwin64\home\username\.cpanm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-ali
en-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      0 [main] perl 9873 child_info_fork::abort: unable to map \??\C:\cygwin64\home\username\.cp
anm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-alien-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0
.dll (using C:\cygwin64\home\username\.cpanm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-ali
en-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      0 [main] perl 9874 child_info_fork::abort: unable to map \??\C:\cygwin64\home\username\.cp
anm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-alien-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0
.dll (using C:\cygwin64\home\username\.cpanm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-ali
en-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      0 [main] perl 9875 child_info_fork::abort: unable to map \??\C:\cygwin64\home\username\.cp
anm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-alien-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0
.dll (using C:\cygwin64\home\username\.cpanm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-ali
en-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126
      0 [main] perl 9876 child_info_fork::abort: unable to map \??\C:\cygwin64\home\username\.cp
anm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-alien-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0
.dll (using C:\cygwin64\home\username\.cpanm\work\1585951500.9548\Alien-Build-2.17\.tmp\test-ali
en-lKpeYG\auto\Test\Alien\XS\Mod0\Mod0.dll), Win32 error 126

**Comment by plicease** [comment]
Fixed in 2.20 🎉
